Debugger/src/Connection/Firebase.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Debugger\Connection;
+
+use Google\Cloud\Core\ArrayTrait;
+use Kreait\Firebase\Factory;
+use Kreait\Firebase\Database;
+
+/**
+ * Implementation of the Firebase connection
+ */
+class Firebase implements ConnectionInterface
+{
+    use ArrayTrait;
+
+    /**
+     * @var Factory
+     */
+    private $factory;
+
+    /**
+     * @var Database
+     */
+    private $database;
+
+    /**
+     * @var array
+     */
+    private $breakpoints;
+
+    public function __construct(array $config = [])
```

Comment:
If we do decide and agree on a new connection class, we'll definitely need some tests for the same.

Reply:
I'll see if I can figure out how to add some tests for this